test/puppetlabs/trapperkeeper/services/status/status_core_test.clj

```diff
@@ -15,4 +15,17 @@
 
       (is (nil? (schema/check ServicesInfo @status-fns)))
       (is (= 2 (count (get @status-fns "foo"))))
-      (is (= 1 (count (get @status-fns "bar")))))))
+      (is (= 1 (count (get @status-fns "bar")))))
+
+    (testing (str "registering a service status callback function with a "
+                  "version that already exists causes an error")
+      (is (thrown? IllegalStateException
+                   (update-status-context status-fns "foo"
+                                          "1.1.0" 2 (fn [] "foo repeat")))))
+
+    (testing (str "registering a service status callback function with a "
+                  "status version that already exists but a different service "
+                  "version causes an error")
+      (is (thrown? IllegalStateException
+                   (update-status-context status-fns "foo"
+                                          "1.2.0" 2 (fn [] "foo repeat")))))))
```

Comment:
I would think that we do want to allow registration for more than one `service-version`? For example, we make some changes to puppet server's status function in puppet server 2.5.0 and thus have a puppet server status function v2, but we still want to support puppet server status function v1. Thus we would have in puppet server's init

```clojure
(register-callback "puppet-server" "2.5.0" 1 puppetserver-status-v1)
(register-callback "puppet-server" "2.5.0" 2 puppetserver-status-v2)
```